### PR TITLE
Aggregation of cosmetic changes made during work on REPL PRs: librustc_mir

### PR DESCRIPTION
**Pull request by alexreg**
Factored out from hacking on rustc for work on the REPL.

r? @Centril

### CODE REVIEW

File: src/librustc_mir/interpret/memory.rs [hunk 1]
```diff
@@ -94,13 +95,14 @@ pub struct Memory<'mir, 'tcx, M: Machine<'mir, 'tcx>> {
...
-    // FIXME: this should not be public, but interning currently needs access to it
+    //
+    // FIXME: this should not be public, but interning currently needs access to it.
     pub(super) dead_alloc_map: FxHashMap<AllocId, (Size, Align)>,
 
     /// Extra data added by the machine.
     pub extra: M::MemoryExtra,
 
-    /// Lets us implement `HasDataLayout`, which is awfully convenient.
+    /// Lets us implement `HasDataLayout`, which is very convenient.
```
Comment (Centril) [inline src/librustc_mir/interpret/memory.rs]:
```suggestion
    /// Lets us implement `HasDataLayout`, which is convenient.
```

File: src/librustc_mir/interpret/operand.rs [hunk 2]
```diff
@@ -26,8 +26,8 @@ pub use rustc::mir::interpret::ScalarMaybeUndef;
 /// operations and fat pointers. This idea was taken from rustc's codegen.
 /// In particular, thanks to `ScalarPair`, arithmetic operations and casts can be entirely
 /// defined on `Immediate`, and do not have to work with a `Place`.
-#[derive(Copy, Clone, Debug, Hash, PartialEq, Eq)]
-pub enum Immediate<Tag=(), Id=AllocId> {
+#[derive(RustcEncodable, RustcDecodable, Copy, Clone, Debug, Hash, PartialEq, Eq)]
```
Comment (Centril) [inline src/librustc_mir/interpret/operand.rs]: Semantic change.

File: src/librustc_mir/interpret/operand.rs [hunk 3]
```diff
@@ -123,8 +123,8 @@ impl<'tcx, Tag> ::std::ops::Deref for ImmTy<'tcx, Tag> {
 /// An `Operand` is the result of computing a `mir::Operand`. It can be immediate,
 /// or still in memory. The latter is an optimization, to delay reading that chunk of
 /// memory and to avoid having to store arbitrary-sized data here.
-#[derive(Copy, Clone, Debug, Hash, PartialEq, Eq)]
-pub enum Operand<Tag=(), Id=AllocId> {
+#[derive(Copy, Clone, Debug, Hash, PartialEq, Eq, RustcEncodable, RustcDecodable)]
```
Comment (Centril) [inline src/librustc_mir/interpret/operand.rs]: Semantic change.

File: src/librustc_mir/interpret/place.rs [hunk 4]
```diff
@@ -17,8 +17,8 @@ use super::{
     RawConst, Immediate, ImmTy, ScalarMaybeUndef, Operand, OpTy, MemoryKind, LocalValue,
 };
 
-#[derive(Copy, Clone, Debug, Hash, PartialEq, Eq)]
-pub struct MemPlace<Tag=(), Id=AllocId> {
+#[derive(Copy, Clone, Debug, Hash, PartialEq, Eq, RustcEncodable, RustcDecodable)]
```
Comment (Centril) [inline src/librustc_mir/interpret/place.rs]: Semantic change.

File: src/librustc_mir/interpret/place.rs [hunk 5]
```diff
@@ -604,29 +604,28 @@ where
...
+                // resolved ID. Also see the doc comment of `Memory::get_static_alloc`.
                 let alloc_id = self.tcx.alloc_map.lock().create_static_alloc(cid.instance.def_id());
                 let ptr = self.tag_static_base_pointer(Pointer::from(alloc_id));
                 MPlaceTy::from_aligned_ptr(ptr, layout)
             }
         })
     }
 
-    /// Computes a place. You should only use this if you intend to write into this
-    /// place; for reading, a more efficient alternative is `eval_place_for_read`.
```
Comment (Centril) [inline src/librustc_mir/interpret/place.rs]: The warning here is presumably there for a reason, revert.

File: src/librustc_mir/interpret/step.rs [hunk 6]
```diff
@@ -92,19 +92,19 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
...
 
-            // Mark locals as dead
+            // Mark locals as dead.
             StorageDead(local) => {
                 let old_val = self.storage_dead(local);
                 self.deallocate_local(old_val)?;
             }
 
-            // No dynamic semantics attached to `FakeRead`; MIR
+            // No dynamic semantics attached to FakeRead; MIR
```
Comment (Centril) [inline src/librustc_mir/interpret/step.rs]: Revert.

File: src/librustc_mir/transform/add_retag.rs [hunk 7]
```diff
@@ -121,7 +121,7 @@ impl<'tcx> MirPass<'tcx> for AddRetag {
                 }
                 TerminatorKind::Drop { .. } |
                 TerminatorKind::DropAndReplace { .. } => {
-                    // `Drop` is also a call, but it doesn't return anything so we are good.
+                    // Drop is also a call, but it doesn't return anything so we are good.
```
Comment (Centril) [inline src/librustc_mir/transform/add_retag.rs]: Revert.

File: src/librustc_mir/transform/generator.rs [hunk 8]
```diff
@@ -269,7 +269,7 @@ impl MutVisitor<'tcx> for TransformVisitor<'tcx> {
     fn visit_basic_block_data(&mut self,
                               block: BasicBlock,
                               data: &mut BasicBlockData<'tcx>) {
-        // Remove StorageLive and StorageDead statements for remapped locals
+        // Remove StorageLive and StorageDead statements for remapped locals.
```
Comment (Centril) [inline src/librustc_mir/transform/generator.rs]:
```suggestion
        // Remove `StorageLive` and `StorageDead` statements for remapped locals.
```

File: src/librustc_mir/transform/qualify_consts.rs [hunk 9]
```diff
@@ -1702,9 +1702,9 @@ impl<'tcx> MirPass<'tcx> for QualifyAndPromoteConstants<'tcx> {
                 Checker::new(tcx, def_id, body, mode).check_const().1
             };
 
-            // In `const` and `static` everything without `StorageDead`
```
Comment (Centril) [inline src/librustc_mir/transform/qualify_consts.rs]: Revert the removal of backticks here in this file.

**Comment by oli-obk** [comment]
Thanks for the PR. Unfortunately, as per our rules set up in #58619, I am closing this PR.

**Pull request closed without merging**